sonic_package_manager/service_creator/creator.py

```diff
@@ -476,28 +495,143 @@ def set_initial_config(self, package):
             cfg = conn.get_config()
             new_cfg = init_cfg.copy()
             utils.deep_update(new_cfg, cfg)
+            self.validate_config(new_cfg)
             conn.mod_config(new_cfg)
 
     def remove_config(self, package):
-        """ Remove configuration based on init-cfg tables, so having
-        init-cfg even with tables without keys might be a good idea.
-        TODO: init-cfg should be validated with yang model
-        TODO: remove config from tables known to yang model
+        """ Remove configuration based on package YANG module.
 
         Args:
             package: Package object remove initial configuration for.
         Returns:
             None
         """
 
-        init_cfg = package.manifest['package']['init-cfg']
-        if not init_cfg:
+        if not package.metadata.yang_module_str:
             return
 
-        for conn in self.sonic_db.get_connectors():
-            for table in init_cfg:
-                for key in init_cfg[table]:
-                    conn.set_entry(table, key, None)
+        module_name = self.cfg_mgmt.get_module_name(package.metadata.yang_module_str)
+        for tablename, module in self.cfg_mgmt.sy.confDbYangMap.items():
+            if module.get('module') != module_name:
+                continue
+
+            for conn in self.sonic_db.get_connectors():
+                keys = conn.get_table(tablename).keys()
+                for key in keys:
+                    conn.set_entry(tablename, key, None)
+
+    def validate_config(self, config):
+        """ Validate configuration through YANG.
+
+        Args:
+            config: Config DB data.
+        Returns:
+            None.
+        Raises:
+            Exception: if config does not pass YANG validation.
+        """
+
+        config = sonic_cfggen.FormatConverter.to_serialized(config)
+        log.debug(f'validating configuration {pformat(config)}')
+        # This will raise exception if configuration is not valid.
+        # NOTE: loadData() modifies the state of ConfigMgmt instance.
+        # This is not desired for configuration validation only purpose.
+        # Although the config loaded into ConfigMgmt instance is not
+        # interesting in this application so we don't care.
+        self.cfg_mgmt.loadData(config)
+
+    def install_yang_module(self, package: Package):
+        """ Install package's yang module in the system.
+
+        Args:
+            package: Package object.
+        Returns:
+            None
+        """
+
+        if not package.metadata.yang_module_str:
+            return
+
```

Comment:
So no indication to the caller on success or not. Is that OK in the usage model?

Reply:
It's ok. If there is no yang module attached in package metadata it is not an error, we just don't install yang model in the system.